js/views/modals/Settings/Advanced.js

```diff
@@ -47,6 +48,35 @@ export default class extends baseVw {
     return super.getFormData(subset);
   }
 
+  clickPurge() {
+    this.purgeCache();
+  }
+
+  /**
+   * Call to the server to remove cached files that are being shared on IPFS.
+   * This call should not be aborted when the view is removed, it's critical the user is informed if
+   * the call fails, even if they have navigated away from the view.
+   */
+  purgeCache() {
+    this.getCachedEl('.js-purge').addClass('processing');
+    this.getCachedEl('.js-purgeComplete').addClass('hide');
+
+    this.purge = $.post(app.getServerUrl('ob/purgecache'))
+      .always(() => {
+        this.getCachedEl('.js-purge').removeClass('processing');
+      })
+      .fail((xhr) => {
+        const failReason = xhr.responseJSON && xhr.responseJSON.reason || '';
+        openSimpleMessage(
+          app.polyglot.t('settings.advancedTab.server.purgeError'),
+          failReason);
+      })
+      .done(() => {
+        this.getCachedEl('.js-purgeComplete').removeClass('hide');
+      });
+  }
+
+
```

Comment:
It might not be a bad idea to pass `isPurging: this.purge && this.purge.state() === 'pending'` into the template and then use that in the template to decide whether the `processing` class should be applied to the button.

Otherwise, if you re-render during a purge (e.g. user clicks Save), then the spinner state is lost and the user might just kick off another purge.